functions.js

```diff
@@ -0,0 +1,181 @@
+/*
+
+filedrag.js - HTML5 File Drag & Drop demonstration
+Featured on SitePoint.com
+Developed by Craig Buckler (@craigbuckler) of OptimalWorks.net
+
+*/
+(function () {
+	// - - - - - - - - - - - - - - - - - - - - - //
+	// HOST AND PORT
+	//HOST = 'neuralmechanics.ai/abscbn-api';
+
+	HOST = 'http://127.0.0.1:5000';
+
+	// URL Calls to flask server
+	URL_ZIP_FILE = HOST + "/zip_file"
+	URL_TRAIN = HOST + "/train"
+	URL_ANALYZE = HOST + "/analyze"
+	// - - - - - - - - - - - - - - - - - - - - - //
+
+	var output_data = [];
+	var foamtree;
+
+	// getElementById
+	function $id(id) {
+		return document.getElementById(id);
+	}
+
+	// output information
+	function output(type, msg) {
+		var m = (type == 'image' ? $id("image_div") : $id('text_div'));
+		m.innerHTML = msg + m.innerHTML;
+	}
+
+	// file drag hover
+	function fileDragHover(e) {
+		e.stopPropagation();
+		e.preventDefault();
+		e.target.className = (e.type == "dragover" ? "hover" : "");
+	}
+
+	// file selection
+	function fileSelectHandler(e) {
+		console.log("You dropped the file!");
+		$id("alert").style = "display: none;";
+
+		// cancel event and hover styling
+		fileDragHover(e);
+
+		// dragged image from HTML page
+		if (typeof e.dataTransfer != 'undefined' && e.dataTransfer.files.length === 0) {
+			console.log('received image from outside!');
+		} else {
+			var files = e.target.files || e.dataTransfer.files;
+
+			// process all File objects
+			for (var i = 0, f; f = files[i]; i++) {
+				console.log(f);
+				fileTypeChecker(f);
+				var encodedFile = getBase64(f);
+
+				parseFile(f);
+			}
+		}
+	}
+
+	function fileTypeChecker(f) {
+		$id("loading").style.display = "block";
+		var fileExtension = f["name"].split(".").pop();
+
+		if ($id("select-input-type").value === "Photos (ZIP File)") {
+			var expectedExtension = "zip";
+		} else if ($id("select-input-type").value === "Videos (MP4, etc.)") {
+			var expectedExtension = "mp4";
+		}
+
+		if (fileExtension === expectedExtension) {
+			console.log("Recieved ZIP file");
+		} else if (
+			fileExtension === expectedExtension ||
+			fileExtension === "mkv" ||
+			fileExtension === "3gp"
+		) {
+			console.log("Received Video");
+		} else {
+			alert("Expected input " + $id("select-input-type").value + ", but received a " + fileExtension + " file.");
+		}
+		return null
+	}
+
+	function getBase64(f) {
+		var reader = new FileReader();
+		reader.readAsDataURL(f);
+		reader.onload = function () {
+			var data = reader.result;
+
+			function callback(output) {
+				console.log(output);
+			}
+			sendRequest(URL_ZIP_FILE, data, callback);
+		};
+		reader.onerror = function (error) {
+			console.log("Error: ", error)
+		};
+	}
+	
+	function train() {
```

Comment:
This function handles the response of the training button, and saves the output to the `<span>` tag with id `accuracy`. See comment on 86 of the `index.html` file.